### DISCUSSION TITLE
Does StrongSwan support tunnel nesting?

### DISCUSSION THREAD

**Discussion by daniel0720** [strongswan/strongswan#2492] [Q&A]
Hello.
I want to do an experiment about tunnel nesting, but failed because the ISAKMP packet of inner tunnel can't transport by outer tunnel.
Here is my record.
There are 3 PCs in my experiment, the outer IP is 192.168.11.0/24, the inner ip is 10.x.x.x, the same network interface. First I create a site2site tunnel between linux1 and linux2, then I create another site2site  tunnel between linux2 and linux3. Through which the 10.1.0.1  can ping 10.2.0.2.
But when I want to create a host2host tunnel between 10.1.0.1 and 10.2.0.2, the  ISAKMP packet can't encrypted by the tunnel between linux1 and linux2. What I expect is the packets of creating tunnel  between 10.1.0.1 and 10.2.0.2 can encrypted by the tunnel I just created.

Any help  would be greatly appreciated.

Best Regards,

![image](https://github.com/user-attachments/assets/20d08c1e-7646-49ad-9112-05c5def33905)

Next is the config files. OS is ubuntu24.04, StrongSwan version is 5.9.13

Linux1

![image](https://github.com/user-attachments/assets/4dafb9f4-4414-4747-9f4b-f795cbf244ad)
![image](https://github.com/user-attachments/assets/2db0c79d-233f-4edd-933e-7783b1fcba41)

linux2
![image](https://github.com/user-attachments/assets/b8964fdf-6429-41cb-a403-4ef1fe9cb96b)
![image](https://github.com/user-attachments/assets/aace1f28-5750-486c-96df-59f55dac2483)

Linux3
![image](https://github.com/user-attachments/assets/519588be-13e0-453f-8adb-d7c7f7c8d815)
![image](https://github.com/user-attachments/assets/fe83d004-79b5-444e-8007-4435cad1e2c8)

10.1.0.1 ping 10.2.0.2
![image](https://github.com/user-attachments/assets/d60f9ba6-1f5b-49a1-be65-929190a528fd)

inner tunnel initiate
![image](https://github.com/user-attachments/assets/ee52e1ce-f730-4ddb-82a8-15ddf44da1b5)
![image](https://github.com/user-attachments/assets/078c272e-46e1-42c1-b4ae-62dfcc9ab5e4)

Linux1 iptables
![image](https://github.com/user-attachments/assets/0667273e-4ab4-4f10-bb09-d7054b38c560)

**Comment by tobiasbrunner** [comment]
> but failed because the ISAKMP packet of inner tunnel can't transport by outer tunnel.

Yep, IKE traffic is excluded from IPsec processing via socket policies to prevent the packets from getting tunneled if the traffic selectors overlap with the IKE endpoints. While there is an option that enables an alternative approach that uses port-specific bypass policies (`charon.plugins.kernel-netlink.port_bypass`), bypassing IKE traffic currently can't be disabled completely (i.e. you'd have to remove those policies manually after the daemon installed them, e.g. in a start script).

**Reply by daniel0720** [reply]
Thank you very much for the help.
You mean set `charon.plugins.kernel-netlink.port_bypass` to `yes`? What policies should be removed manually?

**Reply by tobiasbrunner** [reply]
Yes, and the policies are those installed by that setting (check `ip xfrm policy`).
